packages/developer-console-spa/src/components/lighthouse/CreateProjectForm.tsx

```diff
@@ -0,0 +1,89 @@
+import { Button, Form, FormGroup, FormHelperText, TextInput } from '@patternfly/react-core';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { createLightHouseProjects } from '../../services/lighthouse';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon';
+
+const CreateProjectForm = ( props: any ) => {
+    const [ projectName, setProjectName ] = useState('');
+    const [ repoURL, setRepoURL ] = useState('');
+    const [ baseBranch, setBaseBranch ] = useState( '' );
+    const [ isPrimaryLoading, setIsPrimaryLoading ] = useState( false );
+    const { register, handleSubmit, watch, formState: { errors } } = useForm( {
+        mode: 'onBlur'
+    } );
+
+    const createProject = () => {
+        setIsPrimaryLoading( true );
+        let project = {
+            name: projectName,
+            baseBranch: baseBranch,
+            externalUrl: repoURL
+        };
+        createLightHouseProjects( project ).then( res => {
+            setIsPrimaryLoading( false );
+            props.setSelectedProject(res);
+            props.setShowConfirmation(true);
+        });
+    }
+    return (
+        <Form onSubmit={ handleSubmit( createProject ) } className="lighthouse-projectform--margintop" >
+            <FormGroup
+                isRequired
+                fieldId=""
+                helperTextInvalid="Project Name is mandatory"
+                helperTextInvalidIcon={ <ExclamationCircleIcon /> }
+                validated={ errors.projectName ? "error" : "default"  }
+                label="Project Name">
+                <TextInput
+                    isRequired
```

Comment:
React-hook-form is a library utilized for form handling, so that we can avoid our focus on the form handling part and  move to bussiness logic. 

I would recommend to use either react-hook-form(preferred) or use our own logic, but not both together as both are trying to solve same thing.

I don't think register can be used like that with patternfly react due to onChange callback difference in native one and patternfly one. Either register can be destructed on top or using of controller component as said in doc.

https://react-hook-form.com/get-started/#IntegratingwithUIlibraries

Reply:
Fixed